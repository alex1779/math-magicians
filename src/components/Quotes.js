function Quotes() {
  return (
    <>
      <p style={{
        fontSize: 20, color: 'black', paddingTop: '10vw', paddingLeft: '10vw',
      }}
      >
        Pure mathematics is, in its way, the poetry of logical ideas.â€” Albert Einstein.
      </p>
    </>
  );
}

export default Quotes;
