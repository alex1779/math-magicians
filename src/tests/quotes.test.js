import React from 'react';
import { waitFor, render } from '@testing-library/react';
import { setupServer } from 'msw/node';
import Quotes from '../components/Quotes';
import handlers from './mocks';

const server = setupServer(...handlers); // Create a server with your handlers

beforeAll(() => server.listen()); // Start the server before running tests
afterEach(() => server.resetHandlers()); // Reset request handlers between tests
afterAll(() => server.close()); // Close the server after all tests

test('Confirm that Quotes renders correctly', async () => {
  const { container } = render(<Quotes />);

  await waitFor(() => {
    // Check that the li element is rendered
    const liElement = container.querySelector('li');
    expect(liElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
