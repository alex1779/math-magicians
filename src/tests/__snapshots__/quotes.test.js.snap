// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm that Quotes renders correctly 1`] = `
<div>
  <ul
    class="quotes"
  >
    <li>
      Money is the root of all evil. -Mark Twain
    </li>
  </ul>
</div>
`;
